base/db/R/insert.format.vars.R

```diff
@@ -0,0 +1,162 @@
+#'  )
+#'   insert.format.vars(con = bety$con, name = "LTER-HFR-103", mimetype_id = 1090, notes = "NPP from Harvard Forest.", header = FALSE, skip = 0, formats_variables = formats_variables_tibble)
+#' }
+insert.format.vars <- function(con, format_name, mimetype_id, notes = NULL, header = TRUE, skip = 0, formats_variables = NULL){
+  
+  # Test if name is a character string
+  if(!is.character(name)){
+    PEcAn.logger::logger.error(
+      "Name must be a character string"
+    )
+  }
+  
+  # Test if format name already exists
+  name_test <- dplyr::tbl(con, "formats") %>% dplyr::select(id, name) %>% dplyr::filter(name %in% format_name) %>% collect()
+  name_test_df <- as.data.frame(name_test)
+  if(!is.null(name_test_df[1,1])){
+    PEcAn.logger::logger.error(
+      "Name already exists"
+    )
+  }
+ 
+  # Test if format name already exists
+  name_test <- dplyr::tbl(con, "formats") %>% dplyr::select(id, name) %>% dplyr::filter(name %in% format_name) %>% collect()
+  name_test_df <- as.data.frame(name_test)
+  if(!is.null(name_test_df[1,1])){
+    PEcAn.logger::logger.error(
+      "Name already exists"
+    )
+  }
+  
+   #Test if skip is an integer
+  if(!is.numeric(skip)){
+  PEcAn.logger::logger.error(
+    "Skip must be of type integer"
+    )
+  }
+  
+  # Test if header is a Boolean
+  if(!is.logical(header)){
+    PEcAn.logger::logger.error(
+      "Header must be of type Boolean"
+    )
+  }
+  
+  # Test if notes are a character string
+  if(!is.character(notes)&!is.null(notes)){
+    PEcAn.logger::logger.error(
+      "Notes must be of type character"
+    )
+  }
+  
+  ######## Formats-Variables tests ###############
+  if(!is.null(formats_variables)){
+    for(i in 1:nrow(formats_variables)){
+      if(!is.numeric(formats_variables[[i,"variable_id"]])){
+        PEcAn.logger::logger.error(
+          "variable_id must be an integer"
+        )
+      }
+      if(!is.character(formats_variables[[i, "name"]])&!is.na(formats_variables[[i, "name"]])){
+        PEcAn.logger::logger.error(
+          "Variable name must be of type character or NA"
+        )
+      }
+      if(!is.character(formats_variables[[i, "unit"]])&!is.na(formats_variables[[i, "unit"]])){
+        PEcAn.logger::logger.error(
+          "Units must be of type character or NA"
+        )
+      }
+      if(!is.character(formats_variables[[i, "storage_type"]])&!is.na(formats_variables[[i, "storage_type"]])){
+        PEcAn.logger::logger.error(
+          "storage_type name must be of type character or NA"
+        )
+      }
+      if(!is.character(formats_variables[[i, "column_number"]])&!is.na(formats_variables[[i, "column_number"]])){
+        PEcAn.logger::logger.error(
+          "column_number name must be of type character or NA"
+        )
+      }
+    }
+  }
+  
+  ## convert NA to "" for inserting into db ##
+   formats_variables[is.na(formats_variables)] <- ""
+   
+  ###  udunit tests ###
+  for(i in 1:nrow(formats_variables)){
```

Comment:
1. This test should probably be within the `if(!is.null(formats_variables)){`
2. it's also better to use `seq_along` over 1:nrow
Both these changes protect against the sequence ending up over 1:0 and then iterating over empty tables.

Reply:
Right. I'll move the tests into that if statement. I don't follow on your second point.